### PR TITLE
Rolling upgrade ubuntu 18 with ldap enabled 2020.1

### PR DESCRIPTION
**Pull request by fgelcer**
## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fgelcer** [comment]
there is job running [here](https://jenkins.scylladb.com/view/enterprise/job/scylla-enterprise/job/rolling-upgrade/job/rolling-upgrade-ubuntu18-test/3/)

**Comment by fgelcer** [comment]
> there is job running [here](https://jenkins.scylladb.com/view/enterprise/job/scylla-enterprise/job/rolling-upgrade/job/rolling-upgrade-ubuntu18-test/3/)

this test failed (still need to investigate and report) when `systemctl start scylla-server.service` failed

**Comment by fgelcer** [comment]
> > there is job running [here](https://jenkins.scylladb.com/view/enterprise/job/scylla-enterprise/job/rolling-upgrade/job/rolling-upgrade-ubuntu18-test/3/)
> 
> this test failed (still need to investigate and report) when `systemctl start scylla-server.service` failed

this is because of https://github.com/scylladb/scylla/issues/7819

**Comment by fgelcer** [comment]
@amoskong , could you please review, while i'm trying to clear out the failure of the test (that seems to be scylladb/scylla#7819)..
we could advance with this PR, and even merge it, as this is a new test, that isn't being triggered other than manuall.

**Review by amoskong — APPROVED** [review]
Looks good.

File: vars/rollingUpgradePipeline.groovy [hunk 1]
```diff
@@ -62,7 +62,7 @@ def call(Map pipelineParams) {
                     script {
                         def tasks = [:]
 
-                        for (version in supportedUpgradeFromVersions(env.GIT_BRANCH, pipelineParams.base_versions)) {
```
Comment (fruch) [inline vars/rollingUpgradePipeline.groovy:65]: can you revert this change ?

Comment (fruch) [reply vars/rollingUpgradePipeline.groovy:65]: we need this filter for OSS versions, so they won't upgrade from enterprise versions

**Review by fruch — CHANGES_REQUESTED** [review]
please put `supportedUpgradeFromVersions` back into place

File: vars/rollingUpgradePipeline.groovy (hunk 1 above)
Comment (fgelcer) [reply vars/rollingUpgradePipeline.groovy:65]: if i do, this test will not work, as the branch name does not include the format of enterprise branches (it is running from master branch). this test is intended to run from enterprise 2020.1 to latest enterprise master.
as @amoskong mentioned, we are much more strict with the verification of the base version vs new version.
so, we will need to improve it, by checking the tags of the image, if there are tags in it, otherwise, we will need to check it after we have a 1st node installed..

**Comment by fgelcer** [comment]
> please put `supportedUpgradeFromVersions` back into place

can't...
https://github.com/scylladb/scylla-cluster-tests/pull/3107#discussion_r555208638

File: vars/rollingUpgradePipeline.groovy (hunk 1 above)
Comment (fruch) [reply vars/rollingUpgradePipeline.groovy:65]: for testing, you can do what you want.
this logic stays until we'll have better logic for the pipeline to select why list of version to start from.

I can't merge this, since it would start upgrades on OSS from 2020.1 -> master, or 2020.1 - > 4.4. 

both you and @amoskong  can thing of a better way todo that, but for now this logic would stays.

Comment (fgelcer) [reply vars/rollingUpgradePipeline.groovy:65]: i understand that, if someone tries to upgrade from enterprise to OSS it should fail, but, how many jobs/builds we have that this mechanism must stay to protect it?
the only time i saw this "failure" was when i was trying to run this, from a branch on my private fork that did not contain the enterprise and its version in the branch name...

> it would start upgrades on OSS from 2020.1 -> master, or 2020.1 - > 4.4

something or someone must trigger that, to have that "wrong" test running.. or our infra has autonomy to run rolling upgrades by itself?
i really want to understand the logic behind this mechanism today (in the past, when the procedures were less strict, maybe, it made sense, but now...)

Comment (roydahan) [reply vars/rollingUpgradePipeline.groovy:65]: For now, we don't need tis change because i don't really care about enterprise branch.
We can have the job directly in 2020.1, we need to test upgrades from 2020.1.x to 2020.1.y.

Comment (fgelcer) [reply vars/rollingUpgradePipeline.groovy]: done.

**Comment by fgelcer** [comment]
@roydahan / @fruch , done

**Review by roydahan — APPROVED** [review]

**Comment by roydahan** [comment]
backported, please configure the jobs and give them a try.

**Comment by fgelcer** [comment]
> backported, please configure the jobs and give them a try.

doing it right now

**Comment by fgelcer** [comment]
> backported, please configure the jobs and give them a try.

also, this one must be backported #3043 as it was not yet..
